libs/langchain/langchain/schema/vectorstore.py

```diff
@@ -553,6 +553,7 @@ class VectorStoreRetriever(BaseRetriever):
         "similarity",
         "similarity_score_threshold",
         "mmr",
+        "similarity_score"
```

Comment:
Let's not introduce a new search type for this purpose -- the names are too similar sounding and will be confusing; e.g., why does similarity_score return a score but not similarity_score_threshold?

Instead, let's add a parameter called `score_key: Optional[str] = None` If specified, it'll place a score when available in the metadata under the specified key. It sounds like this is only supported for now for `similarity_score_threshold` so we should document that clearly and raise a ValueError on an invalid parameter combination.